app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,436 @@
+    }, [selectActiveChannels]);
+
+    const searchChannels = useCallback((text: string) => {
+        if (text) {
+            const active = selectActiveChannels();
+            const filtered = filterChannelsByTerm(active, text);
+            setTerm(text);
+            if (
+                filtered.length !== visibleChannels.length ||
+                filtered.reduce((shouldUpdate, c, i) => shouldUpdate || c.id !== visibleChannels[i].id, false)
+            ) {
+                setVisibleChannels(filtered);
+            }
+        } else {
+            stopSearch();
+        }
+    }, [selectActiveChannels, visibleChannels, stopSearch]);
+
+    const onPressDropdownElement = useCallback((channelType: string) => {
+        setTypeOfChannels(channelType);
+    }, []);
+
+    const renderLoading = useCallback(() => {
+        return (
+            <Loading
+                containerStyle={style.loadingContainer}
+                style={style.loading}
+                color={theme.buttonBg}
+            />
+        );
+
+    //Style is covered by the theme
+    }, [theme]);
+
+    useEffect(() => {
+        mounted.current = true;
+        doGetChannels();
+        return () => {
+            mounted.current = false;
+        };
+    }, []);
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case CLOSE_BUTTON_ID:
+                        close();
+                        break;
+                    case CREATE_BUTTON_ID: {
+                        // TODO part of https://mattermost.atlassian.net/browse/MM-39733
+                        // Update this to use the proper constant and the proper props.
+                        const screen = 'CreateChannel';
+                        const title = intl.formatMessage({id: 'mobile.create_channel.public', defaultMessage: 'New Public Channel'});
+                        const passProps = {
+                            channelType: General.OPEN_CHANNEL,
+                            categoryId,
+                        };
+
+                        // Go to or show modal?
+                        goToScreen(screen, title, passProps);
+                        break;
+                    }
+                }
+            },
+        }, componentId);
+        return () => {
+            unsubscribe.remove();
+        };
+    }, [intl.locale, categoryId]);
+
+    useDidUpdate(() => {
+        doGetChannels();
+
+        // Since doGetChannels also depends on typeOfChannels, it should be up to date when reaching this effect.
+    }, [typeOfChannels]);
+
+    useDidUpdate(() => {
+        const active = selectActiveChannels();
+        if (term) {
+            setVisibleChannels(filterChannelsByTerm(active, term));
+        } else {
+            setVisibleChannels(active);
+        }
+    }, [selectActiveChannels]);
```

Comment:
this would depend on `term`, right?

Reply:
Changes in term are handled by the value passed to `<SearchBar>` prop `onChangeText`, `searchChannels`. This effect is just to update the visible channels when the active channels change.